src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java

```diff
@@ -30,48 +34,128 @@
-            if (((EntityLivingBase) target).hurtTime != 0 && wait.getValue()) continue;
-            if (!walls.getValue() && (!mc.player.canEntityBeSeen(target) && !canEntityFeetBeSeen(target)))
+            if (!EntityUtil.isLiving(target)) {
+                continue;
+            }
+            if (target == mc.player) {
+                continue;
+            }
+            if (mc.player.getDistance(target) > range.getValue()) {
+                continue;
+            }
+            if (((EntityLivingBase) target).getHealth() <= 0) {
+                continue;
+            }
+            if (((EntityLivingBase) target).hurtTime != 0 && wait.getValue()) {
+                continue;
+            }
+            if (!walls.getValue() && (!mc.player.canEntityBeSeen(target) && !canEntityFeetBeSeen(target))) {
                 continue; // If walls is on & you can't see the feet or head of the target, skip. 2 raytraces needed
+            }
             if (players.getValue() && target instanceof EntityPlayer && !Friends.isFriend(target.getName())) {
                 attack(target);
                 return;
             } else {
                 if (EntityUtil.isPassive(target) ? animals.getValue() : (EntityUtil.isMobAggressive(target) && mobs.getValue())) {
-                    if (ModuleManager.isModuleEnabled("AutoTool")) AutoTool.equipBestWeapon();
+                    if (ModuleManager.isModuleEnabled("AutoTool")) {
+                        AutoTool.equipBestWeapon();
+                    }
                     attack(target);
                     return;
                 }
             }
         }
     }
 
+    private boolean checkSharpness(ItemStack stack) {
+
+        if (stack.getTagCompound() == null) {
+            return false;
+        }
+
+        NBTTagList enchants = (NBTTagList) stack.getTagCompound().getTag("ench");
+		
+		if (enchants == null) {
+			return false;
+		}
+			
+
+        for (int i = 0; i < enchants.tagCount(); i++) {
+            NBTTagCompound enchant = ((NBTTagList) enchants).getCompoundTagAt(i);
+            if (enchant.getInteger("id") == 16) {
+                int lvl = enchant.getInteger("lvl");
+                if (lvl >= 16) {
+                    return true;
+                }
+                break;
+            }
+        }
+
+        return false;
+
+    }
+
     private void attack(Entity e) {
+
+        if (sharpness.getValue()) {
+
+            if (!checkSharpness(mc.player.getHeldItemMainhand())) {
+
+                int newSlot = -1;
+
+                for (int i = 0; i < 9; i++) {
+                    ItemStack stack = mc.player.inventory.getStackInSlot(i);
+                    if (stack == ItemStack.EMPTY) {
+                        continue;
+                    }
+                    if (checkSharpness(stack)) {
+                        newSlot = i;
+                        break;
+                    }
+                }
+
+                if (newSlot == -1) {
+                    return;
```

Comment:
Should we really not attack at all when we don't have a 32k? Players probably don't want to keep switching the 32k setting on and off, so having aura function like usual (until a 32k is found) might be a better idea.

Reply:
A lot of Players use Kami in Addition to Future for example, and use Kamis Aura for 32k Swords only, cause its attacking quite fast. I set it up this way cause i used it myself like this.
Of course i can change it.

Reply:
I'd rather have you change it. We can't throw people without future under the bus - and honestly, Kami should be designed with it being the 'dominant client' in mind.

Reply:
Fixed.